raft/raft_test.go

```diff
@@ -1911,6 +1911,244 @@ func TestCommitAfterRemoveNode(t *testing.T) {
 	}
 }
 
+func TestNormalLeaderTransfer(t *testing.T) {
```

Comment:
I would suggest to add a comment to make it clear that we test the code path that the transferee has the most up-to-date log entires when transfer starts.